kalite/distributed/features/steps/superuser_create.py

```diff
@@ -0,0 +1,145 @@
+from behave import *
+from selenium.common.exceptions import NoSuchElementException
+
+from django.core.urlresolvers import reverse
+
+from kalite.testing.behave_helpers import *
+
+from kalite.testing.mixins.facility_mixins import CreateFacilityMixin, CreateStudentMixin
+
+
+from django.contrib.auth.models import User
+
+@given("there is superuser")
+def step_impl(context):
+    if User.objects.exists():
+        pass
+    else:
+        User.objects.create_superuser(username='superusername', password='superpassword', email='super@email.com')
+
+@then("there should be no modal displayed")
+def step_impl(context):
+    assert not find_id_with_wait(context, "superusercreate-container")
+
+@given("there is no superuser")
+def step_impl(context):
+    if User.objects.exists():
+        User.objects.all().delete()
+
+@then("I should see a modal")
+def step_impl(context):
+    assert find_id_with_wait(context, "superusercreate-container")
+
+@given("the username is empty")
+def step_impl(context):
+    fill_username(context, "")
+
+@when('I click the create button')
+def impl(context):
+    create_button = find_css_class_with_wait(context, "create-btn")
+    create_button.click()
```

Comment:
What's the expected outcome of the button click? A page load? A style change? There is a potential race condition if in a subsequent step you don't add a wait for the expected outcome.

If there's more than one possible outcome (like a success or failure) I recommend waiting for one or the other to occur and then asserting the desired outcome, in order to differentiate between success, failure, and unexpected outcomes like timeouts in which neither success nor failure occurs.